javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,373 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+class LinkedList[E]() extends AbstractSequentialList[E]
+    with List[E] with Deque[E] with Cloneable with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  import LinkedList._
+
+  private var head: Node[E] = null
+  private var last: Node[E] = head
+
+  private var _size = 0
+
+  def getFirst(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekFirst()
+
+  def getLast(): E =
+    if (_size <= 0) throw new NoSuchElementException()
+    else peekLast()
+
+  def removeFirst(): E = {
+    if (isEmpty()) throw new NoSuchElementException()
+    val ret = head.value
+
+    val oldHead = head
+    head = oldHead.next
+
+    if (head ne null)
+      head.prev = null
+    else
+      head = last
+
+    _size -= 1
+    ret
```

Comment:
Similarly to what you do in `removeNode()`, you could return `oldHead.value` here instead of `ret`, which means you don't need `ret` in the first place.